File: app/src/main/java/com/semanienterprise/softcom/ui/util/GUISpinner.java
package com.semanienterprise.softcom.ui.util;

import android.content.Context;
import android.util.AttributeSet;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.LinearLayout;
import android.widget.Spinner;
import android.widget.TextView;

import com.semanienterprise.softcom.SpinnerInterface;

public class GUISpinner extends LinearLayout {
    TextView label;
    ArrayAdapter<String> aa;
    Spinner spinner;
    SpinnerInterface spinnerInterface;

    public GUISpinner(Context context, String labelText, String options) {
        super(context);
        spinnerInterface = (SpinnerInterface)context;
        label = new TextView(context);
        label.setText(labelText);
        spinner = new Spinner(context);
        String[] opts = options.split("\\|");
        aa = new ArrayAdapter<>(context, android.R.layout.simple_spinner_item, opts);
        spinner.setAdapter(aa);
        spinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                spinnerInterface.spinnerOptionClicked(spinner.getSelectedItem().toString());
            }

            @Override
            public void onNothingSelected(AdapterView<?> parent) {
            }
        });
        this.addView(label);
        this.addView(spinner);
    }

    public GUISpinner(Context context, AttributeSet attrs) {
        super(context, attrs);
        // TODO Auto-generated constructor stub
    }

    public String getValue() {
        return spinner.getSelectedItem().toString();
    }
}

File: app/src/main/java/com/semanienterprise/softcom/ui/util/GUISpinner.kt
package com.semanienterprise.softcom.ui.util

import android.content.Context
import android.view.View
import android.widget.*
import com.semanienterprise.softcom.SpinnerInterface

class GUISpinner(context: Context, labelText: String, options: String) : LinearLayout(context) {
    private var label: TextView = TextView(context)
    private var aa: ArrayAdapter<String>
    internal var spinner: Spinner
    internal var spinnerInterface: SpinnerInterface = context as SpinnerInterface

    init {
        label.text = labelText
        spinner = Spinner(context)
        val opts = options.split("\\|".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        aa = ArrayAdapter(context, android.R.layout.simple_spinner_item, opts)
        spinner.adapter = aa

        spinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
            override fun onItemSelected(parent: AdapterView<*>, view: View, position: Int, id: Long) {
                spinnerInterface.spinnerOptionClicked(spinner.selectedItem.toString())
            }

            override fun onNothingSelected(parent: AdapterView<*>) {}
        }
        this.addView(label)
        this.addView(spinner)
    }

    val value: String
        get() = spinner.selectedItem.toString()
}
